playbooks/docker-machine-functional-devstack-mitaka/run.yaml

```diff
@@ -0,0 +1,59 @@
+- hosts: all
+  become: yes
+  roles:
+    - clone-devstack-gate-to-workspace
+    - create-devstack-local-conf
+    - install-devstack
+    - config-golang
+  tasks:
+    - name: Create Docker machine
+      shell:
+        cmd: |
+          set -ex
+          set -o pipefail
+          git clone https://github.com/sstephenson/bats.git
+          pushd bats
+          ./install.sh /usr/local
+          popd
+          # install Docker
+          apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D || true
+          apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D || true
+          apt-add-repository 'deb http://apt.dockerproject.org/repo ubuntu-trusty main'
+          apt-get update
+          apt-cache policy docker-engine
+          apt-get install -y docker-engine=1.12.6-0~ubuntu-trusty
+          cat /lib/systemd/system/docker.service
+          sed -r -i "s|(ExecStart)=(.+)|\1=\2 --iptables=false|" /lib/systemd/system/docker.service
+          cat /lib/systemd/system/docker.service
+          service docker restart
+          ifconfig -a
+
+          curl -O http://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img
+          openstack image create --file xenial-server-cloudimg-amd64-disk1.img --min-disk 1 --container-format bare \
+              --disk-format qcow2 "Ubuntu 16.04 Server 64bit"
+
+          mkdir -p $GOPATH/src/github.com/docker/machine
+          cp -r '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}' $GOPATH/src/github.com/docker/
+
+          pushd $GOPATH/src/github.com/docker/machine
+          make build
+          cp ./bin/docker-machine /usr/local/bin/
+          docker-machine version
+
+          export OS_FLAVOR_ID=11
+          export OS_IMAGE_NAME="Ubuntu 16.04 Server 64bit"
+          export OS_DOMAIN_ID=default
+          export OS_SSH_USER=ubuntu
+          export OS_NETWORK_NAME=private
+          export OS_FLOATINGIP_POOL=public
+          openstack security group rule create --proto tcp --dst-port 22 default
```

Comment:
Need to do some command options compatible for mitaka and other releases so that the job can be used for all OpenStack releases.

Reply:
as this job can  only run with Mitaka, because of the auth url issue. maybe this is not a generic job definition for other OpenStack releases.  Also, the docker packages are specific for `ubuntu-trusty`

Reply:
We should not need to install specific docker version, let package manage tools to handle these, just install like
```
apt-get install -y docker-engine
```